homeassistant/components/hlk_sw16/__init__.py

```diff
@@ -57,84 +54,113 @@
 
 
 async def async_setup(hass, config):
+    """Component setup, do nothing."""
+    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in config:
+        return True
+
+    for device_id in config[DOMAIN]:
+        conf = config[DOMAIN][device_id]
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data={CONF_HOST: conf[CONF_HOST], CONF_PORT: conf[CONF_PORT]},
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass, entry):
     """Set up the HLK-SW16 switch."""
     # Allow platform to specify function to register new unknown devices
```

Comment:
```suggestion
        hass.data.setdefault(DOMAIN, {})   
```

Last thing, need to move this back to `async_setup_entry` since it can be poped when the last config entry is unloaded.

Reply:
added/moved back